packages/manager/apps/web-office/src/pages/dashboard/users/ModalEditUsers.component.tsx

```diff
@@ -0,0 +1,258 @@
+import { UserParamsType } from '@/api/api.type';
+
+export default function ModalEditUsers() {
+  const { t } = useTranslation(['dashboard/users/edit', 'common']);
+  const navigate = useNavigate();
+
+  const { serviceName: selectedServiceName } = useParams();
+  const [searchParams] = useSearchParams();
+  const activationEmail = searchParams.get('activationEmail');
+  const licencePrepaidName = searchParams.get('licencePrepaidName');
+
+  const { addError, addSuccess } = useNotifications();
+
+  const goBackUrl = useGenerateUrl('..', 'path');
+  const onClose = () => navigate(goBackUrl);
+
+  const {
+    data: userDetail,
+    isLoading: isUserDetailLoading,
+  } = useOfficeUserDetail(activationEmail, licencePrepaidName);
+
+  const [isLoading, setIsLoading] = useState(
+    !userDetail || isUserDetailLoading,
+  );
+
+  const {
+    control,
+    setValue,
+    getValues,
+    formState: { isDirty, isValid, errors },
+  } = useForm({
+    defaultValues: {
+      firstname: userDetail?.firstName || '',
+      lastname: userDetail?.lastName || '',
+      login: userDetail?.activationEmail.split('@')[0] || '',
+    },
+    mode: 'onChange',
+    resolver: zodResolver(EDIT_USERS_FORM_SCHEMA),
+  });
+
+  useEffect(() => {
+    if (userDetail && !isUserDetailLoading) {
+      setValue('firstname', userDetail.firstName);
+      setValue('lastname', userDetail.lastName);
+      setValue('login', userDetail.activationEmail.split('@')[0]);
+      setIsLoading(false);
+    }
+  }, [userDetail, isUserDetailLoading]);
+
+  const { mutate: editUser, isPending: isSending } = useMutation({
+    mutationFn: (params: UserParamsType) => {
+      return licencePrepaidName
+        ? putOfficeLicenseDetails(licencePrepaidName, params)
+        : putOfficeUserDetail(selectedServiceName, activationEmail, params);
+    },
+    onSuccess: () => {
+      addSuccess(
+        <OdsText preset={ODS_TEXT_PRESET.paragraph}>
+          {t('dashboard_users_edit_message_success')}
+        </OdsText>,
+        true,
+      );
+    },
+    onError: (error: ApiError) => {
+      addError(
+        <OdsText preset={ODS_TEXT_PRESET.paragraph}>
+          {t('dashboard_users_edit_message_error', {
+            error: error.response?.data?.message,
+          })}
+        </OdsText>,
+        true,
+      );
+    },
+    onSettled: () => {
+      queryClient.invalidateQueries({
+        queryKey: [
+          licencePrepaidName
+            ? getOfficeLicenseQueryKey(selectedServiceName)
+            : getOfficeUsersQueryKey(selectedServiceName),
+        ],
+      });
+      onClose();
+    },
+  });
+
+  const handleSaveClick = () => {
+    const { firstname, lastname, login } = getValues();
+    editUser({
+      firstName: firstname,
+      lastName: lastname,
+      activationEmail: `${login}@${userDetail.activationEmail.split('@')[1]}`,
+    });
+  };
+
+  const handleCancelClick = () => {
+    onClose();
+  };
+
+  return (
+    <form>
```

Comment:
the form can be inside the modal, seems better to me even if it's basically the same result

Reply:
It's fixed